language/tools/move-cli/src/main.rs

```diff
@@ -170,8 +185,9 @@ fn publish(args: &Move, files: &[String]) -> Result<OnDiskStateView> {
     if args.verbose {
         println!("Compiling Move modules...")
     }
-    let interface_dir = interface_files_dir(&args.build_dir)?;
-    let (_, compiled_units) = move_lang::move_compile(files, &[interface_dir], None, None)?;
+    let mut deps = args.mode.prepare_source_files(&args.package_dir)?;
```

Comment:
I think this might be a bit easier if we make this a function for the `Move` struct, so it can just be `args.mode_source_files()`

Reply:
Yup, I thought about that too. Will make the changes.